base/random.jl

```diff
@@ -1203,7 +1203,7 @@ let Floats = Union{Float16,Float32,Float64}
             $randfun{T<:$Floats}(::Type{T}) = $randfun(GLOBAL_RNG, T)
 
             # filling arrays
-            function $randfun!{T}(rng::AbstractRNG, A::AbstractArray{T})
+            function $randfun!{T<:$Floats}(rng::AbstractRNG, A::AbstractArray{T})
```

Comment:
I don't like this either, because it will screw up `randn!` for `Complex` arrays.

I think the existing definition is the right one: `randn!` for an array calls `randn` for each element.  If the latter isn't defined, it throws a `MethodError`.